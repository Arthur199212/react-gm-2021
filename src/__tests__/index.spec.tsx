import { NextPageContext } from 'next'
import React from 'react'
import IndexPage, { getStaticProps } from '@app/pages/index'
import { render } from '@app/tests/testing-utils'

describe('Index Page', () => {
  it('should render properly', async () => {
    const { asFragment } = render(<IndexPage />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('getStaticProps method works properly', async () => {
    const end = jest.fn()
    const writeHead = jest.fn()
    const ctxMock = { res: { end, writeHead } } as unknown
    const staticProps = getStaticProps(ctxMock as NextPageContext)

    expect(staticProps.props).toEqual({})
    expect(writeHead).toBeCalledWith(302, { Location: '/search' })
    expect(end).toBeCalled()
  })
})
