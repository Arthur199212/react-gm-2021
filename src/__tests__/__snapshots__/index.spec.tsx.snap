// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page should render properly 1`] = `<DocumentFragment />`;
